### DISCUSSION TITLE
useLoaderData typescript infer problem

### DISCUSSION THREAD

**Discussion by rlesniak** [remix-run/remix#7331] [Q&A]
My loader return something like that:
```
return json({
  currentDate: new Date()
})
```

My `useLoaderData<typeof loader>` returns following type: `{ currentDate: Date }` 

But in reality, loader value is ISO string.

How can I handle right TS type?

Remix I use: 1.19.3

**Comment by quietCoder500** [comment]
Would [this](https://stackoverflow.com/questions/38359563/how-to-handle-iso-date-strings-in-typescript) help you at all?

**Reply by rlesniak** [reply]
No, problem is with any non-primitive obj
